### PR TITLE
Support Pydantic v2

### LINKED ISSUE #118: Migrate to pydantic v2
See https://docs.pydantic.dev/latest/migration/

Some methods/attrs have new names.This will likely break Scitacean in a few places. I'm not sure if we can just drop support for v1 at this point. So we may need to wrap calls to those methods to support both the new and old names.

### PR DESCRIPTION
**Pull request by jl-wynen**
Fixes #118

This PR makes Scitacean compatible with Pydantic version 2 while maintaining compatibility with 1.*

I did not update the version in `requirements/*` because `autodoc-pydantic` does not yet support v2. So for the time being, the CI and docs builds run with v1.
I tested the changes locally with 2.0 and 1.10.

### CODE REVIEW

File: src/scitacean/testing/backend/seed.py [hunk 1]
```diff
@@ -93,7 +93,7 @@
         techniques=[UploadTechnique(pid="S", name="shoes")],
     ),
     "partially-broken": model.construct(
-        UploadRawDataset,
+        UploadDerivedDataset,
```
Comment (SimonHeybrock) [inline src/scitacean/testing/backend/seed.py:96]: Why this change? Seems unrelated?

**Review by SimonHeybrock — COMMENTED** [review]
Looks ok to me (not familiar with pydantic details).

As you mentioned in the PR message, I do not see any test/CI setup that would ensure that this keeps working for both pydantic versions. Should at least set a reminder to address this soon?

File: src/scitacean/testing/backend/seed.py (hunk 1 above)
Comment (jl-wynen) [reply src/scitacean/testing/backend/seed.py:96]: The type was already 'derived' (see the `type` field). This change fixes a mistake that went under the radar because the old pydantic was more lenient with the types of models. TBH, I don't know why this code didn't raise an exception before but it did now before this change.

**Comment by jl-wynen** [comment]
> Should at least set a reminder to address this soon?

Agree. I could set it up to test with both versions and at least one python version.

**Review by SimonHeybrock — APPROVED** [review]
